projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java

```diff
@@ -98,17 +105,24 @@ public void synchronizeDataJobs() {
     Map<String, ActualDataJobDeployment> actualDataJobDeploymentsFromDBMap =
         deploymentService.findAllActualDataJobDeployments();
 
+    CountDownLatch countDownLatch = new CountDownLatch(dataJobsFromDB.size());
+
     dataJobsFromDB.forEach(
         dataJob ->
-            taskExecutor.execute(
+            executeTask(
                 () ->
                     synchronizeDataJob(
                         dataJob,
                         desiredDataJobDeploymentsFromDBMap.get(dataJob.getName()),
                         actualDataJobDeploymentsFromDBMap.get(dataJob.getName()),
-                        finalDataJobDeploymentNamesFromKubernetes.contains(dataJob.getName()))));
+                        finalDataJobDeploymentNamesFromKubernetes.contains(dataJob.getName())),
+                countDownLatch));
 
-    taskExecutor.shutdown();
+    try {
+      countDownLatch.await();
```

Comment:
Consider adding a few more debug logs. As this could take some time, it might be benefital to know we are waiting.

Reply:
I will add it, but I'm not sure how beneficial it will be, as the update deployment method logs messages, and this one won't be noticeable.